cardano-cli/src/Cardano/CLI/EraBased/Options/Genesis.hs

```diff
@@ -217,18 +218,18 @@ pGenesisCreateStaked sbe envCli =
   pRelayJsonFp =
     parseFilePath "relay-specification-file" "JSON file that specifies the relays of each stake pool."
 
-pGenesisCreateTestNetData :: ShelleyBasedEra era -> EnvCli -> Parser (GenesisCmds era)
-pGenesisCreateTestNetData sbe envCli =
+pGenesisCreateTestNetData :: Exp.Era era -> EnvCli -> Parser (GenesisCmds era)
+pGenesisCreateTestNetData era envCli =
   fmap GenesisCreateTestNetData $
-    GenesisCreateTestNetDataCmdArgs sbe
+    GenesisCreateTestNetDataCmdArgs era
       <$> optional (pSpecFile "shelley")
       <*> optional (pSpecFile "alonzo")
       <*> optional (pSpecFile "conway")
       <*> pNumGenesisKeys
       <*> pNumPools
       <*> pNumStakeDelegs
-      <*> pNumCommittee
-      <*> pNumDReps
+      <*> (case era of Exp.BabbageEra -> pure 0; Exp.ConwayEra -> pNumCommittee) -- Committee doesn't exist in babbage
+      <*> (case era of Exp.BabbageEra -> pure $ DRepCredentials OnDisk 0; Exp.ConwayEra -> pNumDReps) -- DReps don't exist in babbage
```

Comment:
Those inlined cases are not very readable, can you move them to a values in `where`, and just reference them from here?

Reply:
I'd also use `Ord (Some Era)` instead of pattern matching on each era. I mean:
```haskell
if someEra < Some Exp.ConwayEra 
  then pure 0 
  else pNumCommittee
```
and
```haskell
if someEra < Some Exp.ConwayEra
  then pure $ DRepCredentials OnDisk 0
  else Exp.ConwayEra -> pNumDReps
```
Rationale: less maintenance burden when introducing new eras - you won't have to change each usage site of `case`.